### PR TITLE
batches: update README

### PR DESCRIPTION
**Pull request by BolajiOlajide**
### Test plan

Docs update.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
 ```sh
-go run main.go
+go get github.com/sourcegraph/go-diff-patch
 ```
 
-A `generated.patch` file should be generated and work is still ongoing to make it `git-compatible` so it can be applied with `git apply <path/to/patch>`
\ No newline at end of file
+### Usage
+
+The library exports a function `GeneratePatch` which returns the different between two files in a string format. This patch can then be applied to a valid git repository containing the specified file(s).
```
Comment (courier-new) [inline README.md]: Do we intend to only return the diff in a string format?

Maybe we could make the inputs and outputs clearer:

```suggestion
The library exports a function `GeneratePatch` which computes edits between the original and updated file contents and produces a unified diff for them as a string. This patch can then be applied to a valid git repository containing the specified file(s).
```

**Review by courier-new — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (BolajiOlajide) [reply README.md]: Yes, at this moment that's all we do., however in the future we can easily extend this to any other format we like.

File: main.go [hunk 2]
```diff
@@ -8,13 +8,15 @@ import (
...
-// TODO: add description
+// GeneratePatch generates a unified diff that is git-compatible and highlights the difference
+// between originalContent and updatedContent.
 func GeneratePatch(filename, originalContent, updatedContent string) string {
 	uri := span.URI(filename)
 	diffs, err := myers.ComputeEdits(uri, originalContent, updatedContent)
 	if err != nil {
-		// Ideally, this would never happen because the error returned here is always nil.
-		// Error is returned to satisfy the interface defined in `diff/diff.go#L24`
+		// Ideally, this would never happen because the error returned here is always nil for
```
Comment (Piszmog) [inline main.go]: I think from a wording perspective, we should not use "Ideally". This conveys to me that this *may* happen one day. If this is the case, maybe a `panic` is not the right solution. If an error can never actually happen (for reason you state), then `panic` is a good use.

```suggestion
		// An error can never be returned here. It is always nil. The returned error is to satisfy the interface defined in `internal/diff/diff.go#L24`.
```

File: examples/example.go [hunk 3]
```diff
@@ -8,7 +8,7 @@ import (
 
 func main() {
 	patch := godiffpatch.GeneratePatch("test.txt", original, updated)
-	err := os.WriteFile("gitignore.patch", []byte(patch), 0644)
+	err := os.WriteFile("test.txt", []byte(patch), 0644)
```
Comment (Piszmog) [inline examples/example.go:11]: Is it any prettier just to print the difference or is a file easier to read?

**Review by Piszmog — APPROVED** [review]
Looks good to me other than the nits.

Also, the only other nit I have is the name of the file `main.go`. This is a bit strange for a library that does not have a `main` function to have that file. When I first opened the repo, I was confused as I thought this was a library and not a binary. I think a different name for `main.go` would be better to prevent confusion from others. Maybe just `go_diff_patch.go` or something like that.

File: examples/example.go (hunk 3 above)
Comment (BolajiOlajide) [reply examples/example.go:11]: It was easier for me to read from a file when experimenting initially so the logic stayed. I'm not opposed to printing it though.

**Pull request merged**